### DISCUSSION TITLE
macos 15.2 ptrdiff_t bug

### DISCUSSION THREAD

**Discussion by cxwx** [clangd/clangd#2265] [Q&A] [answered]
Hi all, after upgrade to lastest macos 15.2,
the clangd server met problem of ptrdiff_t.
Does anyone have this bugs?
![截屏2024-12-23 15 08 37](https://github.com/user-attachments/assets/075730d5-f12b-40e7-951c-ab26d6ac49ef)
![截屏2024-12-23 15 11 12](https://github.com/user-attachments/assets/50439524-8bd4-4cc9-bec2-c59917d64b72)

**Comment by kammce** [comment]
I think I'm getting the same error. Try running this command:

```bash
clangd --enable-config --check=src/main.cpp --query-driver="**"
```

And take the compilation command and try running it and see if it works. If it doesn't work remove this include from it:

```bash
-isystem /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include 
```

and see if it compiles. If so, then that include is messing up your compilation process. I don't have help to get rid of it, but it would be nice if more people had this issue.

**Comment by cxwx** [comment, marked as answer]
below solved this problem. Cannot use the lib in /Library now
```yml
    # "-I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/c++/v1/",
    # "-I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/",
    # 总是有一堆 BUG，靠CLion 辅助解决。
    "-I/Applications/X/C/D/P/M/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/",
    "-I/Applications/X/C/D/P/M/Developer/SDKs/MacOSX.sdk/usr/include/",
```
